src/ascii/port.rs

```diff
@@ -1205,14 +1333,67 @@ impl<'a, B: Backend> Port<'a, B> {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn poll_until<C, F>(&mut self, cmd: C, mut predicate: F) -> Result<Reply, AsciiError>
+    pub fn poll_until<C, F>(&mut self, cmd: C, predicate: F) -> Result<Reply, AsciiError>
```

Comment:
Need to add/update tests for this change in behaviour and the new `poll_until_*` methods.

Reply:
I have moved the changes to `poll_until*` methods to a separate PR.